linux_pipeline/Jenkinsfile_kernel_bisect_commit_validator

```diff
@@ -1,435 +1,669 @@
+              i='0'
+              IFS=$'\t\n'
+              for url in $(cat developer_patches); do
+                  i=$(( i + 1 ))
+                  protocol=${url%%://*}
+                  url=${url#*//}
+                  test_user=${url%%*@*}
+                  if [[ $protocol == "scp" ]] && [[ $test_user != "" ]]; then
+                      sed -i "${i}s/.*/scp:\\/\\/$SCP_USERNAME_DEFAULT@${url//\\//\\\\/}/" developer_patches
+                  fi
+              done
+              IFS=$' '
+              bash build_artifacts.sh \\
+                  --git_url "${KERNEL_GIT_URL}" \\
+                  --git_branch "${KERNEL_GIT_BRANCH}" \\
+                  --destination_path "${BUILD_NUMBER}-${BRANCH_NAME}-${KERNEL_ARTIFACTS_PATH}" \\
+                  --install_deps "True" \\
+                  --thread_number "x3" \\
+                  --debian_os_version "16" \\
+                  --build_path "${KERNEL_BUILD_DIR}" \\
+                  --kernel_config "Microsoft/config-azure" \\
+                  --clean_env "False" \\
+                  --use_ccache "True" \\
+                  --use_kernel_folder_prefix "True" \\
+                  --patch_file "developer_patches" \\
+                  --create_changelog "False" \\
+                  --light_build "True" \\
+                  --custom_build_tag="${CUSTOM_BUILD_TAG}"
+              popd
+              '''
+    }
+    sh '''#!/bin/bash
+      echo ${BUILD_NUMBER}-$(crudini --get scripts/package_building/kernel_versions.ini KERNEL_BUILT folder) > ./build_name
+    '''
+    script {
+      currentBuild.displayName = readFile "./build_name"
+    }
+    stash includes: 'scripts/package_building/kernel_versions.ini', name: 'kernel_version_ini'
+    dir ("scripts/package_building") {
+        stash includes: "${env.BUILD_NUMBER}-${env.BRANCH_NAME}-${env.KERNEL_ARTIFACTS_PATH}/**/deb/**", name: env.KERNEL_ARTIFACTS_PATH
+    }
+    sh '''
+        set -xe
+        rm -rf "scripts/package_building/${BUILD_NUMBER}-${BRANCH_NAME}-${KERNEL_ARTIFACTS_PATH}"
+        '''
+    archiveArtifacts 'scripts/package_building/kernel_versions.ini'
+}
 
 
-// this is the main logic
-build_kernel(osDistro)
+def runLisa(platform, secrets, testName, testCategory, testArea, testTag, customTestParams, testIterations) {
+    def customKernel = ""
+    def osVHD = ""
+    def region = params.LISAV2_AZURE_REGION
+    def vmSize = ""
+    def customTestIterations = ""
 
-if (isBootTest) {
-    node ('meta_slave') {
-        boot_test()
+    prepareEnv(env.LISAV2_BRANCH, env.LISAV2_REMOTE)
+    try {
+        unstashKernel(env.KERNEL_ARTIFACTS_PATH)
+        customKernel = "-CustomKernel 'localfile:./${env.BUILD_NUMBER}-${env.BRANCH_NAME}-${env.KERNEL_ARTIFACTS_PATH}/*/deb/*.deb'"
+    } catch (Exception e) {
+        echo "No stash ${env.KERNEL_ARTIFACTS_PATH} found"
     }
-}
 
-if (isPerf) {
-    node (perfServer) {
-        perf()
+    if (testName.equals(env.LISAV2_AZURE_STRESS_TEST)) {
```

Comment:
maybe add a parameter in this method for the vmsize, and check if its null or not

Reply:
this runLisa method became too big and has to many params, will have to break it down somehow